README.md

```diff
@@ -0,0 +1,262 @@
+         |                           |
+         |                           |
+         |     +-------------+       |
+         |     |             |       |
+         +----->     spi     <-------+
+               |             |
+               +-------------+
+
+```
+### Pluggable Components Architecture
+Server has some components like Authentication Manager, Authorization Manager, Cryptography Manager depend on SPI interfaces for which different implementations can be plugged. Below is the diagram which shows how these managers and SPI are involved in normal HTTP request flow. `server` component provides simple implementations of all the SPIs.
+
+```text
+HTTP Request                                                                                   
+     |                                                                     
+     v                                                                     
++--------------------+     +------------------+                            
+| Authentication     |     | Authentication   |                            
+| Manager            |---->| SPI              |                            
++--------------------+     +------------------+                            
+     |                                                                     
+     v                                                                     
++--------------------+     +------------------+                            
+| Authorization      |     | Authorization    |                            
+| Manager            |---->| SPI              |                            
++--------------------+     +------------------+                            
+     |                                                                     
+     v                                                 +------------------+
++--------------------+      +--------------------+     | Cryptography     |
+| Core Logic         |----->| Cryptography       |---->| SPI              |
++--------------------+      +--------------------+     +------------------+
+     |                                                                     
+     v                                                                     
++--------------------+     +------------------+                            
+| Storage            |     | Repository       |                            
+| Repositories       |---->| SPIs             |                            
++--------------------+     +------------------+                            
+```
+
+### Clients SDK
+The Java SDK provides a robust, production-ready client with:
+
+- **Automatic Retries** - Exponential backoff for transient failures
+- **Connection Pooling** - Efficient HTTP connection management
+- **Metrics Integration** - Micrometer and JMX support
+- **Structured Logging** - MDC context for request tracing
+
+See the [detailed client documentation](clients/java/README.md) for advanced usage.
```

Comment:
this section should be after the section explaining clients folder structure just below.